src/Build/Evaluation/Expander.cs

```diff
@@ -1123,13 +1123,23 @@ evaluator._metadata is IItemTypeDefinition itemMetadata &&
                             "ItemReferencingSelfInTarget", itemMetadata.ItemType, metadataName);
                     }
 
-                    if (IsTruncationEnabled(evaluator._options) && metadataValue.Length > CharacterLimitPerExpansion)
+                    if (IsTruncationEnabled(_options) && metadataValue.Length > CharacterLimitPerExpansion)
                     {
-                        metadataValue =
 #if NET
-                            $"{metadataValue.AsSpan(0, CharacterLimitPerExpansion - 3)}...";
+                        metadataValue = string.Concat(metadataValue.AsSpan(0, CharacterLimitPerExpansion - 3), "...");
 #else
-                            $"{metadataValue.Substring(0, CharacterLimitPerExpansion - 3)}...";
+                        // PERF: We need the formatted, truncated string. Using something like a StringBuilder avoids
+                        // needing to use an unsafe block, but this is more efficient.
+                        string truncatedMetadataValue = metadataValue.Substring(0, CharacterLimitPerExpansion);
+                        unsafe
+                        {
+                            fixed (char* truncatedMetadataPointer = truncatedMetadataValue)
+                            {
+                                Span<char> destination = new Span<char>(truncatedMetadataPointer, truncatedMetadataValue.Length);
+                                "...".AsSpan().CopyTo(destination.Slice(CharacterLimitPerExpansion - 3));
+                                metadataValue = truncatedMetadataValue;
+                            }
+                        }
 #endif
```

Comment:
There's another clone of this in the file, should we extract the fast-concat method and use it in both places?